pubsub/google/cloud/pubsub_v1/subscriber/_protocol/streaming_pull_manager.py

```diff
@@ -80,6 +80,20 @@ def _wrap_callback_errors(callback, on_callback_error, message):
         on_callback_error(exc)
 
 
+# TODO
+# check if dropping behavior in Leaser is going to be a problem. (maybe disbale
+# for ordering keys). check if what i'm doing here is okay?
+# investigate other places where Drop is happening. Drop is dangerous because it
+# can happen without the user intending an ack or nack.
+
+# 1) dont drop ordering keys messages OR
+# 2) drop all messages from the ordering key (better user experience)
+#
+# if message is in hold, then lease expires, then you wait for server to send
```

Comment:
This only works if you can be guaranteed that the user callback has been run for all messages you drop or you haven't yet sent the mod ack, but it looks like the mod ack is sent immediately upon receiving the message.

Reply:
These were old todos so this comment isn't relevant anymore. Sorry for the merge-induced confusion!